core/src/main/java/org/geysermc/geyser/command/defaults/ConnectionTestCommand.java

```diff
@@ -25,110 +25,110 @@
 
 package org.geysermc.geyser.command.defaults;
 
+import cloud.commandframework.Command;
+import cloud.commandframework.CommandManager;
+import cloud.commandframework.arguments.standard.IntegerArgument;
+import cloud.commandframework.arguments.standard.StringArgument;
+import cloud.commandframework.context.CommandContext;
+import cloud.commandframework.CommandManager;
+import cloud.commandframework.arguments.standard.IntegerArgument;
+import cloud.commandframework.arguments.standard.StringArgument;
 import com.fasterxml.jackson.databind.JsonNode;
 import org.geysermc.geyser.GeyserImpl;
+import org.geysermc.geyser.api.util.TriState;
 import org.geysermc.geyser.api.util.PlatformType;
 import org.geysermc.geyser.command.GeyserCommand;
 import org.geysermc.geyser.command.GeyserCommandSource;
-import org.geysermc.geyser.session.GeyserSession;
-import org.geysermc.geyser.text.GeyserLocale;
 import org.geysermc.geyser.util.LoopbackUtil;
 import org.geysermc.geyser.util.WebUtils;
-import org.jetbrains.annotations.Nullable;
 
+import java.util.Objects;
 import java.util.Random;
 import java.util.concurrent.CompletableFuture;
 
 public class ConnectionTestCommand extends GeyserCommand {
+
     /*
      * The MOTD is temporarily changed during the connection test.
      * This allows us to check if we are pinging the correct Geyser instance
      */
     public static String CONNECTION_TEST_MOTD = null;
 
-    private final GeyserImpl geyser;
+    private static final String ADDRESS = "address";
+    private static final String PORT = "port";
 
+    private final GeyserImpl geyser;
     private final Random random = new Random();
 
     public ConnectionTestCommand(GeyserImpl geyser, String name, String description, String permission) {
-        super(name, description, permission);
+        super(name, description, permission, TriState.NOT_SET);
         this.geyser = geyser;
     }
 
     @Override
-    public void execute(@Nullable GeyserSession session, GeyserCommandSource sender, String[] args) {
-        // Only allow the console to create dumps on Geyser Standalone
-        if (!sender.isConsole() && geyser.getPlatformType() == PlatformType.STANDALONE) {
-            sender.sendMessage(GeyserLocale.getPlayerLocaleString("geyser.bootstrap.command.permission_fail", sender.locale()));
-            return;
-        }
+    public void register(CommandManager<GeyserCommandSource> manager) {
+        manager.command(baseBuilder(manager)
+            .argument(StringArgument.of(ADDRESS))
+            .argument(IntegerArgument.<GeyserCommandSource>builder(PORT)
```

Comment:
Does this mean that the geyser connectiontest command would now have two instead of one argument?
If yes, we should change the connectiontest command usage - it currently instructs separating the args with `:`

Reply:
Precisely.